### PR TITLE
Shrink error span on @type errors for signatures

### PR DESCRIPTION
**Pull request by sandersn**
Previously, the error span was too large on @type errors on functions
when the type was not a function. The span covered the entire tag. This
PR changes the error node just to be the type of the type tag. In other words,
the error span was previously this:

```
@type {IncorrectType}
```

But is now just this:

```
IncorrectType
```

Fixes the first error from #41974, but not the other two.

Thanks @ashyamanning and @neuhelos!

### CODE REVIEW

**Review by andrewbranch — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
This error still is a little strange because it doesn't actually highlight the function declaration itself. You'd think it would with a "the signature is declared here.

Still an overall improvement.

**Comment by sandersn** [comment]
@DanielRosenwasser I agree that's an improvement -- and this error message is specific to the `@type` mismatch, so it's possible to reword this to be better. It's certainly not clear as-is. I'll add your suggestion to the original bug, since it's still open.
